transfer/src/main/java/com/kin/ecosystem/transfer/AccountInfoService.java

```diff
@@ -0,0 +1,167 @@
+package com.kin.ecosystem.transfer;
+
+import android.app.IntentService;
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.content.ComponentName;
+import android.content.Intent;
+import android.content.pm.ResolveInfo;
+import android.graphics.Color;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.Nullable;
+import android.support.annotation.RequiresApi;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.content.FileProvider;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+
+public abstract class AccountInfoService extends IntentService {
+    public interface AccountInfoListener {
+        void onAccountInfoReady(String data);
+
+        void onAccountInfoFailed(String reason);
+    }
+
+    private static String FILE_PROVIDER_DIR_NAME = "kin_account_info";
+    private static String FILE_NAME = "kinAccountInfo.txt";
+    private static String ACTION_KIN_TRANSFER_INFO = "kin.transfer.info";
+    private static String EXTRA_PACKAGE_KEY = "EXTRA_PACKAGE_KEY";
+    private static String EXTRA_ACTION_KEY = "EXTRA_ACTION_KEY";
+    private static String EXTRA_SERVICE_FULL_PATH_KEY = "EXTRA_SERVICE_FULL_PATH_KEY";
+    private static String FILE_PROVIDER_NAME = "AccountInfoFileProvider";
+    private static String CHANNEL_NAME = "kin transfer";
+    private static String CHANNEL_ID = "0";
+    private static String NOTIFICATION_TRANSFER = "transfer kin";
+
+
+    private String fileFullPath, fileDirPath;
+    private File file;
+
+    public AccountInfoService(String name) {
+        super(name);
+    }
+
+    public abstract void getAccountInfo(AccountInfoListener listener);
+
+    @Override
+    protected void onHandleIntent(final @Nullable Intent intent) {
+        startForeground();
+
+        if (intent != null && intent.getAction().equals(ACTION_KIN_TRANSFER_INFO) && intent.hasExtra(EXTRA_ACTION_KEY)
+                && intent.hasExtra(EXTRA_PACKAGE_KEY) && intent.hasExtra(EXTRA_SERVICE_FULL_PATH_KEY)) {
+            getAccountInfo(new AccountInfoListener() {
+                @Override
+                public void onAccountInfoReady(String data) {
```

Comment:
we need to check it's a valid address (length etc.)?

Reply:
im not sure that this is the job of this module 
the one that provide the info need to check that its valid
if the length is fine but the address doesnt exist

Reply:
👍